### PR TITLE
Add new granttype for reset_password

### PR DESCRIPTION
**Pull request by tej99**
Added a new grantType to the tokens endpoint for reseting password, more specifically to check wether the account is locked or not.
https://trello.com/c/8DJShebh/658-defect-60-us012-found-in-release10-app-is-allowing-the-respondent-to-reset-a-password-when-the-account-is-locked

### CODE REVIEW

File: oauth2server/apps/tokens/decorators.py [hunk 1]
```diff
@@ -169,6 +170,18 @@ def _validate_grant_type(request):
                 except KeyError:
                     raise PasswordRequiredException()
 
+        # reset password grant requires username parameter
+        if grant_type == 'reset_password':
+
+            stdlogger.debug("Grant_type reset_password")
+            try:
+                username = request.POST['username']
```
Comment (benjefferies) [inline oauth2server/apps/tokens/decorators.py:178]: Do we only need to handle post requests?

File: oauth2server/apps/tokens/decorators.py [hunk 2]
```diff
@@ -188,6 +201,22 @@ def _validate_grant_type(request):
             except OAuthAuthorizationCode.DoesNotExist:
                 raise AuthorizationCodeNotFoundException()
 
+        if grant_type == 'reset_password':
```
Comment (benjefferies) [inline oauth2server/apps/tokens/decorators.py]: Could this live with the `if grant_type == 'reset_password':` check above?

**Review by benjefferies — CHANGES_REQUESTED** [review]
I think we need some tests here to document the behaviour

**Review by andrewmil — APPROVED** [review]
LGTM, I agree with Ben's second comment though

Tested locally and everything looks right

Edit: Also would be nice to have codecov running on here? Though I'm not sure how it works with Django's testing structure

File: oauth2server/apps/tokens/decorators.py (hunk 1 above)
Comment (tej99) [reply oauth2server/apps/tokens/decorators.py:178]: It handles both in the nested try and except. NGL i didn't put much thought into this just copied the way it currently does this check with other grant types as this service should be getting replaced soon anyway.

**Review by benjefferies — APPROVED** [review]

**Pull request merged**